### ISSUE TITLE
 learning rate decay and loss 

### ISSUE THREAD

**Issue by pangda72** [CN-zdy/MSU_Net#13]
Hi，CN-zdy,
I have some queries regarding your work? Colud you please give me a help,thank you very much!
In this code,I uncomment the part about learning rate decay,my learning rate will go from 0.001 to 0, I don't know why.And the same time ,my epoch is 300,and loss is 0.165,i think it could be high,I don't know why it could be so high?Could you give me a hand?

Looking forward to your reply.Thanks in advance.
Best.

**Comment by CN-zdy** [comment]
Hi, Pangda.
This is because the initial lr variable is set to 0 when I save the learning rate. In addition, we recommend the use of learning rate reduction strategies.
In this experiment, we saved the learning rate of the whole batch.

**Comment by pangda72** [comment]
> Hi, Pangda.
> This is because the initial lr variable is set to 0 when I save the learning rate. In addition, we recommend the use of learning rate reduction strategies.
> In this experiment, we saved the learning rate of the whole batch.

Hi，CN-zdy
I'm still a little confused about learning rates.my lr is 0.001,when i use  learning rate reduction strategies:

self.scheduler = torch.optim.lr_scheduler.StepLR(self.optimizer, step_size=10, gamma=0.1)

It shows that lr  drops every 10 steps,so each epoch to print lr ,it should be smaller,and  change slowly.
But i run 'trian.py',after each epoch,printing the lr shows 0.001.It takes multiple epochs to go down.(It takes about 20 epochs to go down and it value is directly equal to 0).I felt a little confused.Could you please help me to understand it?
Thanks in advance.
![lr](https://user-images.githubusercontent.com/74399643/133019376-7bc83822-7ac6-47eb-8844-5c11b7f3c49b.png)

**Comment by CN-zdy** [comment]
Hi, Pangda.
lr after the 20th epoch is 0.00001 instead of 0.
The 0.0000 is displayed because only four decimal places are displayed.

**Comment by pangda72** [comment]
> Hi, Pangda.
> lr after the 20th epoch is 0.00001 instead of 0.
> The 0.0000 is displayed because only four decimal places are displayed.

Hi,CN-zdy
Thanks for your answer.I get it.But i have another puzzle, Could you  give me a help.Thank you very much!

 I used  the learning rate reduction strategies that you provide in your code, But the five results under my test_result folder were all worse than those without learning rate  reduction  strategies.
1、use learning rate reduction strategies : DC: 0.768;0.779;0.769;0.755;0.765
2、don't use learning rate reduction strategies : DC: 0.799;0.814;0.813;0.808;0806
![lr](https://user-images.githubusercontent.com/74399643/133209808-8a506c6b-b65b-4a25-9f7f-647c3aadc17d.png)
This is the visualized curve of learning rate in the training process of 100 epoch.when at the 20th epoch, the learning rate increased.I don't know what the problem is.Colud you give me a hint,Why do I get worse results using a learning rate decay strategy, and what changes do I need to make.
Hope to hear from you!
Thanks in advance.

**Comment by pangda72** [comment]
> Hi, Pangda.
> lr after the 20th epoch is 0.00001 instead of 0.
> The 0.0000 is displayed because only four decimal places are displayed.

**Comment by CN-zdy** [comment]
Hi,Pangda.
The training epoch of my is 30. So I set step_size to 10.
You can adapt this parameter to your model training.

**Comment by pangda72** [comment]
> Hi,Pangda.
> The training epoch of my is 30. So I set step_size to 10.
> You can adapt this parameter to your model training.

Hi CN-zdy
Thanks for your guidance, I will try.
Best.
